service/src/main/java/bio/terra/workspace/common/utils/MdcHook.java

```diff
@@ -130,9 +158,26 @@ private Map<String, String> deserializeMdc(@Nullable String serializedMdc) {
   }
 
   @Override
-  public HookAction stateTransition(FlightContext context) throws InterruptedException {
+  public HookAction stateTransition(FlightContext context) {
     logger.info(
         "Flight ID {} changed status to {}.", context.getFlightId(), context.getFlightStatus());
     return HookAction.CONTINUE;
   }
+
+  private void addFlightContextToMdc(FlightContext context) {
+    MDC.put(FLIGHT_ID_KEY, context.getFlightId());
+    MDC.put(FLIGHT_CLASS_KEY, context.getFlightClassName());
+  }
+
+  private void addStepContextToMdc(FlightContext context) {
+    MDC.put(FLIGHT_STEP_CLASS_KEY, context.getStepClassName());
```

Comment:
Small nit: the set of steps here and in removeStepContextFromMdc may be captured in a constant to prevent any disagreement between these two methods.

Reply:
Good shout, updated.